### PR TITLE
Support GOG.com installation directories

### LINKED ISSUE #600: Should add check for default path to IWAD in GOG install
Now that the Doom games are available on GOG.com, Chocolate Doom should add a check to the default install path, like the one for Steam.

### PR DESCRIPTION
**Pull request by chungy**
This should close #600.

Thanks to DuClare on IRC for helping me test and get this working.

### CODE REVIEW

File: src/d_iwad.c [hunk 1]
```diff
@@ -142,22 +142,53 @@ static registry_value_t uninstall_values[] =
     },
 };
 
-// Value installed by the Collector's Edition when it is installed
+// Values installed by the GOG.com and Collector's Edition versions
 
-static registry_value_t collectors_edition_value =
+static registry_value_t gogcom_collectors_edition_values[] =
```
Comment (fragglet) [inline src/d_iwad.c]: The name seems confusing. Can we give it a more meaningful name. For example: root_path_registry_keys?

File: src/d_iwad.c [hunk 2]
```diff
@@ -142,22 +142,53 @@ static registry_value_t uninstall_values[] =
...
+        HKEY_LOCAL_MACHINE,
+        SOFTWARE_KEY "\\GOG.com\\Games\\1435848742",
+        "PATH",
+    },
 };
 
 // Subdirectories of the above install path, where IWADs are installed.
 
-static char *collectors_edition_subdirs[] = 
+static char *gogcom_collectors_edition_subdirs[] =
```
Comment (fragglet) [inline src/d_iwad.c]: Same here. root_path_subdirs?

**Comment by fragglet** [comment]
This seems to match the info that Gez posted on Doomworld: http://www.doomworld.com/vb/post/1414883

Can you double check that the data all matches up correctly, as I believe you said you haven't been able to test this.

File: src/d_iwad.c (hunk 1 above)
Comment (chungy) [inline src/d_iwad.c]: I didn't think of a good name, I'll change it

**Comment by chungy** [comment]
Yes, DuClare on IRC actually bought me copies of the game from GOG.com, I've double-checked all of this by installing them in Wine :) (This is the registry key for one of the games: https://gist.github.com/chungy/1168ccdd13012a208460)

**Comment by fragglet** [comment]
Looks good, thanks!

**Pull request merged**